public/app/core/utils/richHistory.ts

```diff
@@ -52,7 +55,8 @@ export function addToRichHistory(
   /* Keep only queries, that are within the selected retention period or that are starred.
    * If no queries, initialize with empty array
    */
-  const queriesToKeep = richHistory.filter((q) => q.ts > retentionPeriodLastTs || q.starred === true) || [];
+  const queriesToKeep =
+    richHistory.filter((q) => q.ts > retentionPeriodLastTs || q.starred === true).splice(0, MAX_HISTORY_ITEMS) || [];
```

Comment:
This way, we always delete the freshly run query - we are not able to add any new query. Shouldn't we delete the last unstarred query in the array? I mean this works as well, just asking why this solution was preferred.

Reply:
Good point, that should be better!

- [ ] rotate unstarred history items when reaching the limit